packages/node/src/plugins/FederationModuleInfoRuntimeModule.ts

```diff
@@ -12,14 +12,44 @@ class FederationModuleInfoRuntimeModule extends RuntimeModule {
   /**
    * @returns {string} runtime code
    */
-  // eslint-disable-next-line max-lines-per-function
   generate() {
     return Template.asString([
       `${RuntimeGlobals.require}.federation = {`,
       Template.indent([`cache: {},`, `remotes: {},`, `moduleInfo: {},`]),
-      `}`,
+      `};`,
+      // `if (!globalThis.__remote_scopes__) {`,
+      Template.indent([
+        '// backward compatible global proxy',
+        `globalThis.__remote_scope__ = new Proxy(globalThis.__remote_scope__ || __webpack_require__.federation, {`,
```

Comment:
The globalThis.__remote_scope__ proxy in FederationModuleInfoRuntimeModule.ts is used to provide backward compatibility for global proxy. It's a JavaScript Proxy object that intercepts and defines custom behavior for fundamental operations (e.g., property lookup, assignment, enumeration, function invocation, etc.) on the globalThis.__remote_scope__ object.

Here's what it does:

- It's created if globalThis.__remote_scope__ doesn't already exist.
- It proxies the federation object of RuntimeGlobals.require.
- It has two handler functions: get and set.

The get function:

- If the property being accessed is [config](file:///Users/bytedance/dev/universe/packages/node/src/plugins/FederationModuleInfoRuntimeModule.ts#28%2C28-28%2C28), it returns the remotes object from the federation object.
- For any other property, it returns the corresponding property from the cache object of the federation object.

The set function:

- If the property being set is _config, it assigns the value to the remotes object of the federation object.
- For any other property, it assigns the value to the corresponding property in the cache object of the federation object.

This proxy allows the federation object to be accessed and manipulated via globalThis.__remote_scope__, providing a consistent interface for managing federation module information.

Refer to the code here: